modules/node-shared/src/test/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalsSuite.scala

```diff
@@ -0,0 +1,260 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+
+import org.tessellation.node.shared.domain.seedlist.HistoricalSnapshotOrdinals._
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+import fs2.data.csv.DecoderError
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object HistoricalSnapshotOrdinalsSuite extends SimpleIOSuite with Checkers {
+
+  type Range = (NonNegLong, NonNegLong)
+
+  val terminatedRangeGen: Gen[(NonNegLong, NonNegLong, String)] = for {
+    a <- Gen.chooseNum(0L, Long.MaxValue).map(NonNegLong.unsafeFrom)
+    b <- Gen.chooseNum[Long](a, Long.MaxValue).map(NonNegLong.unsafeFrom)
+  } yield (a, b, s"$a-$b")
+
+  def reduce(ranges: List[Range]): Option[NonEmptyList[Range]] = {
+    val sortedRanges = ranges.sortBy { case (start, _) => start.value }
+    val mergedRanges = sortedRanges
+      .foldLeft(List.empty[Range]) {
+        case (acc, (currStart, currEnd)) =>
+          acc match {
+            case (prevStart, prevEnd) :: tail if currStart <= prevEnd =>
+              val newEnd = NonNegLong.unsafeFrom(Math.max(prevEnd, currEnd))
+              (prevStart, newEnd) :: tail
+            case _ =>
+              (currStart, currEnd) :: acc
+          }
+      }
+
+    mergedRanges.reverse.toNel
+  }
```

Comment:
1. The two functions have slightly different signatures. In the tests, `Range` is `(NonNegLong, NonNegLong)` whereas it's a `case class` in the actual implementation.
2. In the implementation, all the functions and `case class`es are private except for `optionalHistoricalSnapshotOrdinalsCellDecoder`, `HistoricalSnapshotOrdinals`, and the error `case class`es. That's because they are only used in that file, per my discussion with the team on this at the last sync meeting. So, I cannot access those functions from the tests. Thus, I duplicated the functions I needed for the purposes of testing.

Reply:
Per our offline discussion, the appropriate `Range` functions are now public and this is not needed anymore.